src/java/org/apache/cassandra/io/util/FileHandle.java

```diff
@@ -322,13 +392,14 @@ public Builder withCrcCheckChance(Supplier<Double> crcCheckChanceSupplier)
          */
         public Builder mmapped(boolean mmapped)
         {
-            this.mmapped = mmapped;
+            if (mmapped)
+                withDiskAccessMode(DiskAccessMode.mmap);
```

Comment:
Should this have an `else` for the false condition? Otherwise remove the parameter and have it always switch to mmaped.